src/Map.js

```diff
@@ -73,7 +72,6 @@ export class Map extends L.Evented {
 
     // Accept layer as config object
     addLayer(layer) {
-        const { layerTypes } = this.options
         let newLayer = layer
 
         if (layer.type && layerTypes[layer.type]) {
```

Comment:
The default behaviour is to pass in a Leaflet layer instance, but it will also accept a config object. I've added a check if "newLayer instanceof L.Layer" which all layers should extend. if createLayer returns null, it will not pass this check.